department-of-veterans-affairs/va.gov-cms
# VACMS-7600: Release content continuously during business hours

## Description

Relates to #7600. (or closes?)

* Continuous release event listener
* Also includes some extra metrics so that we can keep an eye on error states

## Testing done

Automated.

## Screenshots


## QA steps

* `drush va-gov:content-release:toggle-continuous` to enable continuous release (it is off by default)
* `drush va-gov:content-release:reset-state` to force the release state to ready.
* `drush va-gov:content-release:request-frontend-build` to queue a release request
* `drush advancedqueue:queue:process content_release` to process the content release queue (which will advance the state to `dispatched`)
* `drush va-gov:content-release:advance-state starting`
* `drush va-gov:content-release:advance-state inprogress`
* `drush va-gov:content-release:advance-state complete`
* At this point, there should be another release request in the queue if you're testing this during business hours. You can check that on the queue item listing page or you can just re-run the queue process command from above (if it says there were items processed, then it worked as expected). If you're outside of business hours and just want to see it work, add a `return true;` to the top of `RunsDuringBusinessHours->isCurrentlyDuringBusinessHours()`.



What needs to be checked to prove this works?  
What needs to be checked to prove it didn't break any related things?  
What variations of circumstances (users, actions, values) need to be checked?  

As user _uid_ with _user_role_
1. Do this
2. Then
3. Then validate Acceptance Criteria from issue

### Definition of Done


### Select Team for PR review



### Is this PR blocked by another PR?


### Does this PR need review from a Product Owner


### CMS user-facing annoucement

Is an announcement needed to let editors know of this change?
